### PR TITLE
DOCSP-39789 - Compatibility

### PR DESCRIPTION
**Pull request by mongoKart**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-39789
Staging - https://preview-mongodbmongokart.gatsbyjs.io/cpp-driver/docsp-39789-compat/compatibility/

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/includes/language-compatibility-table-cxx.rst [hunk 1]
```diff
@@ -8,24 +8,14 @@
...
-     -
-
-   * - <= 3.7
-     -
-     - ✓
-     - ✓
-     - ✓
-     -
 
-   * - 1.x
```
Comment (mongoKart) [inline source/includes/language-compatibility-table-cxx.rst:27]: Tech reviewer: I couldn't find a v1.x driver in either the docs or driver repo. If it needs to be added, let me know.

File: source/includes/mongodb-compatibility-table-cxx.rst [hunk 2]
```diff
@@ -19,7 +19,7 @@
      - MongoDB 2.6
      - MongoDB 2.4
 
-   * - 3.8+
+   * - 3.8 to 3.10
```
Comment (mongoKart) [inline source/includes/mongodb-compatibility-table-cxx.rst:22]: The rows containing the word "to" are two lines long, which makes the table look a little funny. I experimented with "widths" but couldn't fix it. If you know of a way to fix, let me know!

File: source/compatibility.txt [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+---------------------
+
+The following compatibility table specifies the recommended version or versions
+of the {+driver-long+} for use with a specific version of MongoDB.
+
+The first column lists the driver version.
+
+.. sharedinclude:: dbx/lifecycle-schedule-callout.rst
+
+.. sharedinclude:: dbx/compatibility-table-legend.rst
```
Comment (rustagir) [inline source/compatibility.txt:30]: I; compatibility table legend is duplicated
S: remove it from the top of the included file

File: source/includes/mongodb-compatibility-table-cxx.rst (hunk 2 above)
Comment (rustagir) [reply source/includes/mongodb-compatibility-table-cxx.rst:22]: Ah, this is an issue we tried to resolve on some other compat pages to. It's resolved if you reduce the number of overall columns. But that might not be an option here.

**Review by rustagir — APPROVED** [review]
approving with some comments and a bug

File: source/compatibility.txt (hunk 3 above)
Comment (mongoKart) [reply source/compatibility.txt:30]: good catch

File: source/includes/language-compatibility-table-cxx.rst (hunk 1 above)
Comment (kevinAlbs) [reply source/includes/language-compatibility-table-cxx.rst:27]: Versions older than 3.0 correspond to the legacy driver. The legacy driver is [documented as end-of-life](https://mongocxx.org/legacy-v1/). I expect the legacy driver does not need to be mentioned.

**Review by kevinAlbs — APPROVED** [review]

**Pull request merged**